src/Commands/GenericCommand.php

```diff
@@ -1,41 +1,83 @@
 <?php
-
-/*
+/**
  * This file is part of the TelegramBot package.
  *
  * (c) Avtandil Kikabidze aka LONGMAN <akalongman@gmail.com>
  *
  * For the full copyright and license information, please view the LICENSE
  * file that was distributed with this source code.
-*/
+ */
+
 namespace Longman\TelegramBot\Commands;
 
-use Longman\TelegramBot\Request;
 use Longman\TelegramBot\Command;
 use Longman\TelegramBot\Entities\Update;
+use Longman\TelegramBot\Request;
 
+/**
+ * Generic command
+ */
 class GenericCommand extends Command
 {
+    /**
+     * Name
+     *
+     * @var string
+     */
     protected $name = 'Generic';
-    protected $description = 'Handle generic commands or is executed by default when a command is not found';
+
+    /**
+     * Description
+     *
+     * @var string
+     */
+    protected $description = 'Handles generic commands or is executed by default when a command is not found';
+
+    /**
+     * Usage
+     *
+     * @var string
+     */
     protected $usage = '/';
+
+    /**
+     * Version
+     *
+     * @var string
+     */
     protected $version = '1.0.0';
+
+    /**
+     * If this command is enabled
+     *
+     * @var boolean
+     */
     protected $enabled = true;
 
+    /**
+     * Execute command
+     *
+     * @todo This can't be right, as it always returns "Command: xyz not found.. :("
```

Comment:
In my mind (as far as I've understood the different commands so far), there should be 3 different kinds of commands:
- Executed by the library / internal system
- Executed by the Admins
- Executed by the Users (& Admins obviously)

This way, we could separate these different commands more easily, as they have their own specific way of being executed. I see at the moment that many commands do pretty much the same thing, but each one uses lots of identical code, which really should be optimised.

Yes I saw that @ public vs. private. I think public should be the ones all Users can execute and private the ones only Admins can.

I haven't looked deeper into this, but maybe adding an extra layer could help out a lot:
```
-> Command
  |-> AdminCommand
    |-> SendToAllAdminCommand
  |-> UserCommand
    |-> WhoAmIUserCommand
  |-> SystemCommand
    |-> NewChatTitleSystemCommand
```

Something like that, you know what I mean?